google-cloud-pubsub/test/google/cloud/pubsub/subscriber/timed_unary_buffer_test.rb

```diff
@@ -0,0 +1,253 @@
+      rescue => exception
+        raise ::Google::Cloud::Error.from_error(exception)
+      end
+    end
+  
+    subscription.service.mocked_subscriber = stub
+    subscriber = subscription.listen streams: 1 do |msg|
+      msg.acknowledge!
+      called = true
+    end
+  
+    subscriber.start
+  
+    subscriber_retries = 0
+    until called
+      fail "total number of calls were never made" if subscriber_retries > 100
+      subscriber_retries += 1
+      sleep 0.01
+    end
+  
+    sleep 5
+    assert stub.acknowledge_requests.length > 1
+    subscriber.stop
+    subscriber.wait!
+  end
+  
+  it "should call handle error for retry mod ack on retriable error" do
+    pull_res1 = Google::Cloud::PubSub::V1::StreamingPullResponse.new received_messages: [rec_msg1_grpc],
+                                                                    subscription_properties: {
+                                                                        exactly_once_delivery_enabled: true
+                                                                    }   
+    response_groups = [[pull_res1]]
+  
+    stub = StreamingPullStub.new response_groups
+    called = false  
+    def stub.modify_ack_deadline subscription:, ack_ids:, ack_deadline_seconds:
+      @modify_ack_deadline_requests << [subscription, ack_ids.sort, ack_deadline_seconds]
+      begin
+        raise GRPC::InvalidArgument.new "test"
+      rescue => exception
+        raise ::Google::Cloud::Error.from_error(exception)
+      end
+    end
+  
+    subscription.service.mocked_subscriber = stub
+    subscriber = subscription.listen streams: 1 do |msg|
+      msg.modify_ack_deadline! 120
+      called = true
+    end
+  
+    subscriber.start
+  
+    subscriber_retries = 0
+    until called
+      fail "total number of calls were never made" if subscriber_retries > 100
+      subscriber_retries += 1
+      sleep 0.1
+    end
+     
+    sleep 5
+    assert stub.modify_ack_deadline_requests.length > 1
+    subscriber.stop
+    subscriber.wait!
+  end
+  
+  it "should raise other errors on modack" do
+    pull_res1 = Google::Cloud::PubSub::V1::StreamingPullResponse.new received_messages: [rec_msg1_grpc],
+                                                                    subscription_properties: {
+                                                                        exactly_once_delivery_enabled: true
+                                                                    }   
+    response_groups = [[pull_res1]]
+  
+    stub = StreamingPullStub.new response_groups
+    called = false  
+    errors = []
+    def stub.modify_ack_deadline subscription:, ack_ids:, ack_deadline_seconds:
+      raise StandardError.new "Test failure"
+    end
+  
+    subscription.service.mocked_subscriber = stub
+    subscriber = subscription.listen streams: 1 do |msg|
+      msg.modify_ack_deadline! 120
+      called = true
+    end
+
+    subscriber.on_error do |error|
+        errors << error
```

Comment:
Indent 2 spaces. (This also appears several more times below, sometimes on blocks, and sometimes on embedded methods.)